// @mui/x-charts/ChartsSurface.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ChartsSurface = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));
var _system = require("@mui/system");
var React = _interopRequireWildcard(require("react"));
var _useAxisEvents = require("./hooks/useAxisEvents");
var _jsxRuntime = require("react/jsx-runtime");
const _excluded = ["children", "width", "height", "viewBox", "disableAxisListener", "className", "sx"];
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
const ChartChartsSurfaceStyles = (0, _system.styled)('svg', {
  name: 'MuiChartsSurface',
  slot: 'Root'
})(() => ({}));
const ChartsSurface = /*#__PURE__*/React.forwardRef(function ChartsSurface(props, ref) {
  const {
      children,
      width,
      height,
      viewBox,
      disableAxisListener = false,
      sx
    } = props,
    other = (0, _objectWithoutPropertiesLoose2.default)(props, _excluded);
  const svgView = (0, _extends2.default)({
    width,
    height,
    x: 0,
    y: 0
  }, viewBox);
  (0, _useAxisEvents.useAxisEvents)(disableAxisListener);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(ChartChartsSurfaceStyles, (0, _extends2.default)({
    width: width,
    height: height,
    viewBox: `${svgView.x} ${svgView.y} ${svgView.width} ${svgView.height}`,
    ref: ref,
    sx: [{
      '--ChartsLegend-itemWidth': '100px',
      '--ChartsLegend-itemMarkSize': '20px',
      '--ChartsLegend-rootSpacing': '5px',
      '--ChartsLegend-labelSpacing': '5px',
      '--ChartsLegend-rootOffsetY': '-20px'
    }, ...(Array.isArray(sx) ? sx : [sx])]
  }, other, {
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("title", {
      children: props.title
    }), /*#__PURE__*/(0, _jsxRuntime.jsx)("desc", {
      children: props.desc
    }), children]
  }));
});
exports.ChartsSurface = ChartsSurface;
