// @mui/x-charts/legacy/ChartsTooltip/ChartsTooltipTable.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import Box from '@mui/material/Box';
import Paper from '@mui/material/Paper';
import { styled } from '@mui/material/styles';
import { tooltipClasses } from './tooltipClasses';
export var ChartsTooltipPaper = styled(Paper, {
  name: 'MuiChartsTooltip',
  slot: 'Table'
})({});
export var ChartsTooltipTable = styled('table', {
  name: 'MuiChartsTooltip',
  slot: 'Table'
})(function (_ref) {
  var theme = _ref.theme;
  return {
    borderSpacing: 0,
    '& thead td': {
      borderBottom: "solid ".concat(theme.palette.divider, " 1px")
    }
  };
});
export var ChartsTooltipRow = styled('tr', {
  name: 'MuiChartsTooltip',
  slot: 'Row'
})(function (_ref2) {
  var theme = _ref2.theme;
  return {
    'tr:first-of-type& td': {
      paddingTop: theme.spacing(1)
    },
    'tr:last-of-type& td': {
      paddingBottom: theme.spacing(1)
    }
  };
});
export var ChartsTooltipCell = styled('td', {
  name: 'MuiChartsTooltip',
  slot: 'Cell'
})(function (_ref3) {
  var _ref4;
  var theme = _ref3.theme;
  return _ref4 = {
    verticalAlign: 'middle',
    color: theme.palette.text.secondary
  }, _defineProperty(_ref4, "&.".concat(tooltipClasses.labelCell), {
    paddingLeft: theme.spacing(1)
  }), _defineProperty(_ref4, "&.".concat(tooltipClasses.valueCell), {
    paddingLeft: theme.spacing(4),
    color: theme.palette.text.primary
  }), _defineProperty(_ref4, 'td:first-of-type&', {
    paddingLeft: theme.spacing(2)
  }), _defineProperty(_ref4, 'td:last-of-type&', {
    paddingRight: theme.spacing(2)
  }), _ref4;
});
export var ChartsTooltipMark = styled(Box, {
  name: 'MuiChartsTooltip',
  slot: 'Mark'
})(function (_ref5) {
  var theme = _ref5.theme,
    ownerState = _ref5.ownerState;
  return {
    width: theme.spacing(1),
    height: theme.spacing(1),
    borderRadius: '50%',
    backgroundColor: ownerState.color,
    borderColor: theme.palette.background.paper,
    border: "solid ".concat(theme.palette.background.paper, " ").concat(theme.spacing(0.25)),
    boxSizing: 'content-box'
  };
});
