// @mui/x-charts/internals/components/AxisSharedComponents.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ChartsTickLabel = exports.ChartsTick = exports.ChartsLine = exports.ChartsLabel = exports.AxisRoot = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _styles = require("@mui/material/styles");
var _axisClasses = require("../../ChartsAxis/axisClasses");
const AxisRoot = (0, _styles.styled)('g', {
  name: 'MuiChartsAxis',
  slot: 'Root',
  overridesResolver: (props, styles) => styles.root
})({
  [`&.${_axisClasses.axisClasses.directionY}`]: {
    [`.${_axisClasses.axisClasses.tickLabel}`]: {
      dominantBaseline: 'middle'
    },
    [`.${_axisClasses.axisClasses.label}`]: {
      dominantBaseline: 'auto',
      textAnchor: 'middle'
    }
  },
  [`&.${_axisClasses.axisClasses.left}`]: {
    [`.${_axisClasses.axisClasses.tickLabel}`]: {
      dominantBaseline: 'central',
      textAnchor: 'end'
    }
  },
  [`&.${_axisClasses.axisClasses.right}`]: {
    [`.${_axisClasses.axisClasses.tickLabel}`]: {
      dominantBaseline: 'central',
      textAnchor: 'start'
    }
  },
  [`&.${_axisClasses.axisClasses.bottom}`]: {
    [`.${_axisClasses.axisClasses.tickLabel}, .${_axisClasses.axisClasses.label}`]: {
      dominantBaseline: 'hanging',
      textAnchor: 'middle'
    }
  },
  [`&.${_axisClasses.axisClasses.top}`]: {
    [`.${_axisClasses.axisClasses.tickLabel}, .${_axisClasses.axisClasses.label}`]: {
      dominantBaseline: 'baseline',
      textAnchor: 'middle'
    }
  }
});
exports.AxisRoot = AxisRoot;
const ChartsLine = (0, _styles.styled)('line', {
  name: 'MuiChartsAxis',
  slot: 'Line',
  overridesResolver: (props, styles) => styles.line
})(({
  theme
}) => ({
  stroke: theme.palette.text.primary,
  shapeRendering: 'crispEdges',
  strokeWidth: 1
}));
exports.ChartsLine = ChartsLine;
const ChartsTick = (0, _styles.styled)('line', {
  name: 'MuiChartsAxis',
  slot: 'Tick',
  overridesResolver: (props, styles) => styles.tick
})(({
  theme
}) => ({
  stroke: theme.palette.text.primary,
  shapeRendering: 'crispEdges'
}));
exports.ChartsTick = ChartsTick;
const ChartsTickLabel = (0, _styles.styled)('text', {
  name: 'MuiChartsAxis',
  slot: 'TickLabel',
  overridesResolver: (props, styles) => styles.tickLabel
})(({
  theme
}) => (0, _extends2.default)({}, theme.typography.caption, {
  fill: theme.palette.text.primary
}));
exports.ChartsTickLabel = ChartsTickLabel;
const ChartsLabel = (0, _styles.styled)('text', {
  name: 'MuiChartsAxis',
  slot: 'Label',
  overridesResolver: (props, styles) => styles.label
})(({
  theme
}) => (0, _extends2.default)({}, theme.typography.body1, {
  fill: theme.palette.text.primary
}));
exports.ChartsLabel = ChartsLabel;
