// @mui/x-charts/legacy/internals/components/AxisSharedComponents.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
var _$concat, _styled;
import { styled } from '@mui/material/styles';
import { axisClasses } from '../../ChartsAxis/axisClasses';
export var AxisRoot = styled('g', {
  name: 'MuiChartsAxis',
  slot: 'Root',
  overridesResolver: function overridesResolver(props, styles) {
    return styles.root;
  }
})((_styled = {}, _defineProperty(_styled, "&.".concat(axisClasses.directionY), (_$concat = {}, _defineProperty(_$concat, ".".concat(axisClasses.tickLabel), {
  dominantBaseline: 'middle'
}), _defineProperty(_$concat, ".".concat(axisClasses.label), {
  dominantBaseline: 'auto',
  textAnchor: 'middle'
}), _$concat)), _defineProperty(_styled, "&.".concat(axisClasses.left), _defineProperty({}, ".".concat(axisClasses.tickLabel), {
  dominantBaseline: 'central',
  textAnchor: 'end'
})), _defineProperty(_styled, "&.".concat(axisClasses.right), _defineProperty({}, ".".concat(axisClasses.tickLabel), {
  dominantBaseline: 'central',
  textAnchor: 'start'
})), _defineProperty(_styled, "&.".concat(axisClasses.bottom), _defineProperty({}, ".".concat(axisClasses.tickLabel, ", .").concat(axisClasses.label), {
  dominantBaseline: 'hanging',
  textAnchor: 'middle'
})), _defineProperty(_styled, "&.".concat(axisClasses.top), _defineProperty({}, ".".concat(axisClasses.tickLabel, ", .").concat(axisClasses.label), {
  dominantBaseline: 'baseline',
  textAnchor: 'middle'
})), _styled));
export var ChartsLine = styled('line', {
  name: 'MuiChartsAxis',
  slot: 'Line',
  overridesResolver: function overridesResolver(props, styles) {
    return styles.line;
  }
})(function (_ref) {
  var theme = _ref.theme;
  return {
    stroke: theme.palette.text.primary,
    shapeRendering: 'crispEdges',
    strokeWidth: 1
  };
});
export var ChartsTick = styled('line', {
  name: 'MuiChartsAxis',
  slot: 'Tick',
  overridesResolver: function overridesResolver(props, styles) {
    return styles.tick;
  }
})(function (_ref2) {
  var theme = _ref2.theme;
  return {
    stroke: theme.palette.text.primary,
    shapeRendering: 'crispEdges'
  };
});
export var ChartsTickLabel = styled('text', {
  name: 'MuiChartsAxis',
  slot: 'TickLabel',
  overridesResolver: function overridesResolver(props, styles) {
    return styles.tickLabel;
  }
})(function (_ref3) {
  var theme = _ref3.theme;
  return _extends({}, theme.typography.caption, {
    fill: theme.palette.text.primary
  });
});
export var ChartsLabel = styled('text', {
  name: 'MuiChartsAxis',
  slot: 'Label',
  overridesResolver: function overridesResolver(props, styles) {
    return styles.label;
  }
})(function (_ref4) {
  var theme = _ref4.theme;
  return _extends({}, theme.typography.body1, {
    fill: theme.palette.text.primary
  });
});
