// @mui/x-charts/legacy/ChartsSurface.js
import _toConsumableArray from "@babel/runtime/helpers/esm/toConsumableArray";
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["children", "width", "height", "viewBox", "disableAxisListener", "className", "sx"];
import { styled } from '@mui/system';
import * as React from 'react';
import { useAxisEvents } from './hooks/useAxisEvents';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
var ChartChartsSurfaceStyles = styled('svg', {
  name: 'MuiChartsSurface',
  slot: 'Root'
})(function () {
  return {};
});
export var ChartsSurface = /*#__PURE__*/React.forwardRef(function ChartsSurface(props, ref) {
  var children = props.children,
    width = props.width,
    height = props.height,
    viewBox = props.viewBox,
    _props$disableAxisLis = props.disableAxisListener,
    disableAxisListener = _props$disableAxisLis === void 0 ? false : _props$disableAxisLis,
    className = props.className,
    sx = props.sx,
    other = _objectWithoutProperties(props, _excluded);
  var svgView = _extends({
    width: width,
    height: height,
    x: 0,
    y: 0
  }, viewBox);
  useAxisEvents(disableAxisListener);
  return /*#__PURE__*/_jsxs(ChartChartsSurfaceStyles, _extends({
    width: width,
    height: height,
    viewBox: "".concat(svgView.x, " ").concat(svgView.y, " ").concat(svgView.width, " ").concat(svgView.height),
    ref: ref,
    sx: [{
      '--ChartsLegend-itemWidth': '100px',
      '--ChartsLegend-itemMarkSize': '20px',
      '--ChartsLegend-rootSpacing': '5px',
      '--ChartsLegend-labelSpacing': '5px',
      '--ChartsLegend-rootOffsetY': '-20px'
    }].concat(_toConsumableArray(Array.isArray(sx) ? sx : [sx]))
  }, other, {
    children: [/*#__PURE__*/_jsx("title", {
      children: props.title
    }), /*#__PURE__*/_jsx("desc", {
      children: props.desc
    }), children]
  }));
});
