// @mui/x-charts/ChartsTooltip/ChartsTooltipTable.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ChartsTooltipTable = exports.ChartsTooltipRow = exports.ChartsTooltipPaper = exports.ChartsTooltipMark = exports.ChartsTooltipCell = void 0;
var _Box = _interopRequireDefault(require("@mui/material/Box"));
var _Paper = _interopRequireDefault(require("@mui/material/Paper"));
var _styles = require("@mui/material/styles");
var _tooltipClasses = require("./tooltipClasses");
const ChartsTooltipPaper = (0, _styles.styled)(_Paper.default, {
  name: 'MuiChartsTooltip',
  slot: 'Table'
})({});
exports.ChartsTooltipPaper = ChartsTooltipPaper;
const ChartsTooltipTable = (0, _styles.styled)('table', {
  name: 'MuiChartsTooltip',
  slot: 'Table'
})(({
  theme
}) => ({
  borderSpacing: 0,
  '& thead td': {
    borderBottom: `solid ${theme.palette.divider} 1px`
  }
}));
exports.ChartsTooltipTable = ChartsTooltipTable;
const ChartsTooltipRow = (0, _styles.styled)('tr', {
  name: 'MuiChartsTooltip',
  slot: 'Row'
})(({
  theme
}) => ({
  'tr:first-of-type& td': {
    paddingTop: theme.spacing(1)
  },
  'tr:last-of-type& td': {
    paddingBottom: theme.spacing(1)
  }
}));
exports.ChartsTooltipRow = ChartsTooltipRow;
const ChartsTooltipCell = (0, _styles.styled)('td', {
  name: 'MuiChartsTooltip',
  slot: 'Cell'
})(({
  theme
}) => ({
  verticalAlign: 'middle',
  color: theme.palette.text.secondary,
  [`&.${_tooltipClasses.tooltipClasses.labelCell}`]: {
    paddingLeft: theme.spacing(1)
  },
  [`&.${_tooltipClasses.tooltipClasses.valueCell}`]: {
    paddingLeft: theme.spacing(4),
    color: theme.palette.text.primary
  },
  'td:first-of-type&': {
    paddingLeft: theme.spacing(2)
  },
  'td:last-of-type&': {
    paddingRight: theme.spacing(2)
  }
}));
exports.ChartsTooltipCell = ChartsTooltipCell;
const ChartsTooltipMark = (0, _styles.styled)(_Box.default, {
  name: 'MuiChartsTooltip',
  slot: 'Mark'
})(({
  theme,
  ownerState
}) => ({
  width: theme.spacing(1),
  height: theme.spacing(1),
  borderRadius: '50%',
  backgroundColor: ownerState.color,
  borderColor: theme.palette.background.paper,
  border: `solid ${theme.palette.background.paper} ${theme.spacing(0.25)}`,
  boxSizing: 'content-box'
}));
exports.ChartsTooltipMark = ChartsTooltipMark;
